andela/ah-backend-spaces
# #159514071 Users should be able to verify their account

### What does this PR do?
- verifies accounts of registered users 

### Description of Task to be completed?
- Redirecting the users to the front end link after verifying the user 

### How should this be manually tested?
- Clone the repo [Authors Heaven](https://github.com/andela/ah-backend-spaces.git)
--`git clone https://github.com/andela/ah-backend-spaces.git`

- cd into it and create a new [virtual enviroment](https://docs.python.org/3/tutorial/venv.html#creating-virtual-environments)
-- virtualenv venv

- Install project dependencies
--python install -r requirements.txt

- attempt to sign up as a normal user
- Check your email and the information provided to verify your account.

- - You can also try these out in [post man](https://www.getpostman.com/)
 -- make apost request on `https://authorshaven.herokuapp.com/api/users/` using these details
```
{
	"user": {
	    "username": "Williams",
	    "email": "deriwilliams2008@gmail.com",
	    "password": "jakejhjjake@20AA",
	    
	    "callbackurl":"http://yahoo.com"
	}
} 
```


### Any background context you want to provide?
- A front-end URL is passed into the post details as a user is being registered.
- This is the URL that the user shall be redirected to after the verification is completed successfully.

### What are the relevant pivotal tracker stories?
- [#159514071](https://www.pivotaltracker.com/story/show/159514071)

Registering Account
### Screenshots
<img width="1079" alt="screen shot 2018-08-08 at 11 25 20" src="https://user-images.githubusercontent.com/21138053/43825593-d690113c-9afd-11e8-9d46-6e2931240c48.png">

Recieving Email
<img width="1092" alt="screen shot 2018-08-08 at 11 26 20" src="https://user-images.githubusercontent.com/21138053/43825626-ea655258-9afd-11e8-8df4-4b72b29ffd6b.png">

Successfully verified and continueing to callback url
<img width="1379" alt="screen shot 2018-08-08 at 11 26 52" src="https://user-images.githubusercontent.com/21138053/43825740-2eac9098-9afe-11e8-8f17-43a7987ef658.png">
